src/sentry/lang/javascript/symbolicator.py

```diff
@@ -0,0 +1,299 @@
+            url=base_url,
+            project_id=str(project.id),
+            event_id=str(event_id),
+            timeout=settings.SYMBOLICATOR_POLL_TIMEOUT,
+            source=get_internal_source(project, release),
+        )
+
+        self.task_id_cache_key = _task_id_cache_key_for_event(project.id, event_id)
+
+    def _process(self, create_task, task_name):
+        task_id = default_cache.get(self.task_id_cache_key)
+        json_response = None
+
+        with self.sess:
+            try:
+                if task_id:
+                    # Processing has already started and we need to poll
+                    # symbolicator for an update. This in turn may put us back into
+                    # the queue.
+                    json_response = self.sess.query_task(task_id)
+
+                if json_response is None:
+                    # This is a new task, so we compute all request parameters
+                    # (potentially expensive if we need to pull minidumps), and then
+                    # upload all information to symbolicator. It will likely not
+                    # have a response ready immediately, so we start polling after
+                    # some timeout.
+                    json_response = create_task()
+            except ServiceUnavailable:
+                # 503 can indicate that symbolicator is restarting. Wait for a
+                # reboot, then try again. This overrides the default behavior of
+                # retrying after just a second.
+                #
+                # If there is no response attached, it's a connection error.
+                raise RetrySymbolication(retry_after=settings.SYMBOLICATOR_MAX_RETRY_AFTER)
+
+            metrics.incr(
+                "events.symbolicator.response",
+                tags={"response": json_response.get("status") or "null", "task_name": task_name},
+            )
+
+            # Symbolication is still in progress. Bail out and try again
+            # after some timeout. Symbolicator keeps the response for the
+            # first one to poll it.
+            if json_response["status"] == "pending":
+                default_cache.set(
+                    self.task_id_cache_key, json_response["request_id"], REQUEST_CACHE_TIMEOUT
+                )
+                raise RetrySymbolication(retry_after=json_response["retry_after"])
+            else:
+                # Once we arrive here, we are done processing. Clean up the
+                # task id from the cache.
+                default_cache.delete(self.task_id_cache_key)
+                return json_response
+
+    def process_payload(self, stacktraces, dist):
+        return self._process(
+            lambda: self.sess.symbolicate_stacktraces(stacktraces=stacktraces, dist=dist),
+            "symbolicate_stacktraces",
+        )
+
+
+class TaskIdNotFound(Exception):
+    pass
+
+
+class ServiceUnavailable(Exception):
+    pass
+
+
+def get_internal_source(project, release):
```

Comment:
I’m still waiting on your review on https://github.com/getsentry/sentry/pull/43720 where I tried to make all of this more generic so you can easily add a `process_sourcemap` method to symbolicator instead of copy-pasting the whole thing :-)